test/widgets/lightbox_test.dart

```diff
@@ -152,5 +267,148 @@ void main() {
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.2);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.4);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(twentyPercent, 0.0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.6);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.up();
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, FakeVideoPlayerPlatform.kTestVideoDuration * 0.6);
+      check(FakeVideoPlayerPlatform.position).equals(FakeVideoPlayerPlatform.kTestVideoDuration * 0.6);
+
+      // Verify seekTo is called only once.
+      check(FakeVideoPlayerPlatform.callLogs.where((v) => v == 'seekTo').length).equals(1);
+    });
+
+    testWidgets('video advances overtime after dragging the slider', (tester) async {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      const padding = 24.0;
+      final rect = tester.getRect(find.byType(Slider));
+      final trackWidth = rect.width - padding - padding;
+      final trackStartPos = rect.centerLeft + const Offset(padding, 0);
+      final fiftyPercent = trackWidth * 0.5;
+      final halfTime = FakeVideoPlayerPlatform.kTestVideoDuration * 0.5;
+
+      final gesture = await tester.startGesture(trackStartPos);
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, Duration.zero);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.moveBy(Offset(fiftyPercent, 0));
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, halfTime);
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      await gesture.up();
+      FakeVideoPlayerPlatform.pumpEvents();
+      await tester.pump();
+      checkPosition(tester, halfTime);
+
+      // Verify that after dragging ends, video position is at the
+      // halfway point, and after that it starts advancing as the time
+      // passes.
+      check(FakeVideoPlayerPlatform.position).equals(halfTime);
+
+      const waitTime = Duration(seconds: 1);
+      await tester.pump(waitTime);
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, halfTime + (waitTime * 1));
+      check(FakeVideoPlayerPlatform.position).equals(halfTime + (waitTime * 1));
```

Comment:
Would this be effective as a regression test for the potential bug it's meant to catch?

I.e., quoting my [original request](https://github.com/zulip/zulip-flutter/pull/587#discussion_r1597195082):
> a regression test for the bug that the `_isSliderDragging` and `_sliderValue` logic is avoiding as explained in your comment quoted at https://github.com/zulip/zulip-flutter/pull/587#discussion_r1596190776 .

The regression test I suggested there was:
> Drag slider from one point to another. Check the slider appears at the place you dragged it to now, and at every frame until it starts advancing again because the video is playing.

Where "every frame" is key, because the bug you describe in that comment is a single-frame glitch. It seems like skipping over a whole second could bypass such a glitch.

Reply:
While debugging this, I found that it was possible to [correctly mitigate](https://github.com/zulip/zulip-flutter/pull/694/commits/819118f0b57afc024a7de3ad1da641b6d90be0b0) flickering of slider rather than the previous workaround of delaying the redraw when listeners were called.

The first commit applies the correct fix: https://github.com/zulip/zulip-flutter/pull/694/commits/819118f0b57afc024a7de3ad1da641b6d90be0b0

Reply:
Also, making the wait interval lower than 500ms wasn't working because `video_player` plugin runs a `Timer.periodic` with 500ms interval for dispatching position updates via the `ValueNotifier`:
https://github.com/flutter/packages/blob/260102b64c0fac9c66b7574035421fa6c09f5f89/packages/video_player/video_player/lib/video_player.dart#L566

Reply:
> `video_player` plugin runs a `Timer.periodic` with 500ms interval for dispatching position updates

Huh indeed — wacky. That'll work fine when the video is like 10 minutes long so that 500ms is no more than a couple of pixels, but seems annoying for videos that are like 5 seconds long.

And yeah, that behavior is very conspicuous when I look for it. E.g. here on this 12-second video:
https://chat.zulip.org/#narrow/stream/9-issues/topic/Spoiler.20collapse.20.2F.20uncollapse.20is.20unreliable/near/1823188

but really even on this 3-minute video it's noticeable:
https://chat.zulip.org/#narrow/stream/7-test-here/topic/video.20test/near/1776292

Certainly out of scope for this PR, though.